### DISCUSSION TITLE
Does Velox support GS as well as S3?

### DISCUSSION THREAD

**Discussion by sumokiama** [facebookincubator/velox#6164] [Q&A]
I've asked the Gluten people about GS support and they say that it is not yet supported in Velox.
However, in the Velox howto pages, there is a section on settings for GS.

If GS is not yet supported, are there plans to do so?

**Comment by kgpai** [comment]
Apologies can you tell me what GS is ? Is it google cloud storage ?

**Reply by sumokiama** [reply]
Correct. Google cloud storage.

**Reply by kgpai** [reply]
@tigrux Has been adding support for read/write to GCS , with some PRs being merged recently. He can give you a better answer on state of the support.

**Reply by tigrux** [reply]
Hello. Yes, as @kgpai mentioned, we provided early support for GCS. It can read and write uris as gs://bucket:file.
It is not optimized at all, but in our testing it it able to read buckets containing terabytes of data, which has been enough.
We are in the process of adding support for Gluten, though it may take some months for it to be approved and merged.

**Comment by majetideepak** [comment]
The GCS implementation supports both reading and writing. See https://facebookincubator.github.io/velox/configs.html#google-cloud-storage-configuration for the available configuration.

**Reply by sumokiama** [reply]
Yes, we saw the configuration properties but nothing else about GCS. Thank you for verifying!

**Reply by majetideepak** [reply]
There is a small note about GCS here as well https://facebookincubator.github.io/velox/develop/connectors.html#storage-adapters
